controllers/kyma_controller.go

```diff
@@ -146,60 +150,55 @@ func (r *KymaReconciler) CtrlErr(ctx context.Context, kyma *v1alpha1.Kyma, err e
 
 // synchronizeRemote replaces the given pointer to the Kyma Instance with an instance that contains the merged
 // specification of the Control Plane and the Runtime.
-func (r *KymaReconciler) syncRemote(ctx context.Context, syncContext *remote.KymaSynchronizationContext) error {
-	kyma := syncContext.ControlPlaneKyma
-	if kyma.Status.State == v1alpha1.StateDeleting {
-		return nil
-	}
-	remoteKyma, err := syncContext.CreateOrFetchRemoteKyma(ctx)
+func (r *KymaReconciler) syncRemoteKymaSpecAndStatus(
+	ctx context.Context, controlPlaneKyma *v1alpha1.Kyma,
+) error {
+	syncContext := remote.SyncContextFromContext(ctx)
+
+	remoteKyma, err := syncContext.CreateOrFetchRemoteKyma(ctx, controlPlaneKyma)
 	if err != nil {
 		return fmt.Errorf("could not create or fetch remote kyma: %w", err)
 	}
-	if err := syncContext.SynchronizeRemoteKyma(ctx, remoteKyma); err != nil {
+	if err := syncContext.SynchronizeRemoteKyma(ctx, controlPlaneKyma, remoteKyma); err != nil {
 		return fmt.Errorf("sync run failure: %w", err)
 	}
-	syncContext.ReplaceWithVirtualKyma(kyma, remoteKyma)
+	syncContext.ReplaceWithVirtualKyma(controlPlaneKyma, remoteKyma)
 
 	return nil
 }
 
-func (r *KymaReconciler) syncModuleCatalog(ctx context.Context, syncContext *remote.KymaSynchronizationContext) error {
-	if !syncContext.ControlPlaneKyma.Spec.Sync.ModuleCatalog {
-		return nil
+func (r *KymaReconciler) syncModuleCatalog(ctx context.Context, kyma *v1alpha1.Kyma) (bool, error) {
+	if !kyma.Spec.Sync.Enabled || !kyma.Spec.Sync.ModuleCatalog {
```

Comment:
The issue of validating sync fields has been created here https://github.com/kyma-project/lifecycle-manager/issues/320